lib/logstash/outputs/sumologic.rb

```diff
@@ -17,35 +18,98 @@ class LogStash::Outputs::SumoLogic < LogStash::Outputs::Base
   include LogStash::PluginMixins::HttpClient
   
   config_name "sumologic"
-  
+
+  CONTENT_TYPE = "Content-Type"
+  CONTENT_TYPE_LOG = "text/plain"
+  CONTENT_TYPE_GRAPHITE = "application/vnd.sumologic.graphite"
+  CONTENT_TYPE_CARBON2 = "application/vnd.sumologic.carbon2"
+  CATEGORY_HEADER = "X-Sumo-Category"
+  HOST_HEADER = "X-Sumo-Host"
+  NAME_HEADER = "X-Sumo-Name"
+  CLIENT_HEADER = "X-Sumo-Client"
+  TIMESTAMP_FIELD = "@timestamp"
+  METRICS_NAME_PLACEHOLDER = "*"
+  GRAPHITE = "graphite"
+  CARBON2 = "carbon2"
+  CONTENT_ENCODING = "Content-Encoding"
+  DEFLATE = "deflate"
+  GZIP = "gzip"
+  ALWAYS_EXCLUDED = [ "@timestamp", "@version" ]
+
   # The URL to send logs to. This should be given when creating a HTTP Source
   # on Sumo Logic web app. See http://help.sumologic.com/Send_Data/Sources/HTTP_Source
   config :url, :validate => :string, :required => true
 
+  # Define the source category metadata
+  config :source_category, :validate => :string
+
+  # Define the source host metadata
+  config :source_host, :validate => :string
+
+  # Define the source name metadata
+  config :source_name, :validate => :string
+
   # Include extra HTTP headers on request if needed 
-  config :extra_headers, :validate => :hash, :default => []
+  config :extra_headers, :validate => :hash
 
-  # The formatter of message, by default is message with timestamp and host as prefix
-  # use %{@json} tag to send whole event
-  config :format, :validate => :string, :default => "%{@timestamp} %{host} %{message}"
+  # Compress the payload 
+  config :compress, :validate => :boolean, :default => false
+
+  # The encoding method of compress
+  config :compress_encoding, :validate =>:string, :default => DEFLATE
 
   # Hold messages for at least (x) seconds as a pile; 0 means sending every events immediately  
   config :interval, :validate => :number, :default => 0
 
-  # Compress the payload 
-  config :compress, :validate => :boolean, :default => false
+  # The formatter of log message, by default is message with timestamp and host as prefix
+  # Use %{@json} tag to send whole event
+  config :format, :validate => :string, :default => "%{@timestamp} %{host} %{message}"
 
-  # This lets you choose the structure and parts of the event that are sent in @json tag.
+  # Override the structure of @json tag with the given key value pairs
   config :json_mapping, :validate => :hash
+  
+  # Send metric(s) if configured. This is a hash with k as metric name and v as metric value
+  # Both metric names and values support dynamic strings like %{host}
```

Comment:
metric value supports text?

Reply:
use `%{var}` for using the value of variable. the value should be a number, or it will be ignored